mlflow/spark/autologging.py

```diff
@@ -225,13 +225,26 @@ def _notify(self, path, version, data_format):
         """
         if autologging_is_disabled(FLAVOR_NAME):
             return
-        # If there's an active run, simply set the tag on it
+        # If there are active runs, simply set the tag on the latest active run
         # Note that there's a TOCTOU race condition here - active_run() here can actually throw
         # if the main thread happens to end the run & pop from the active run stack after we check
         # the stack size but before we peek
-        active_run = mlflow.active_run()
-        if active_run:
-            _set_run_tag_async(active_run.info.run_id, path, version, data_format)
+
+        # Note Spark datasource autologging is hard to support thread-local behavior,
+        # because the spark event listener callback (jvm side) does not have the python caller
+        # thread information, so set the tag to the latest active run ignoring threading
+        # information, this way keeps the consistent behavior with released MLflow.
+        latest_active_run = None
+        for active_run_stack in _active_run_stack._value_dict.values():
```

Comment:
Should we guard against the active run stack being empty for some reason? 

``` python
try:
    latest_active_run = None
    for active_run_stack in _active_run_stack._value_dict.values():
        for active_run in active_run_stack:
            if (
                latest_active_run is None
                or active_run.start_time > latest_active_run.start_time
            ):
                latest_active_run = active_run
except Exception:
    # If an exception occurs (e.g., empty run stack or access issues), set to None
    latest_active_run = None
```

Reply:
no need, it can't be None. Its initial value must be `[]` , see https://github.com/mlflow/mlflow/blob/fed898436a558cd8678d3b0b29986e25cb0f8cbe/mlflow/tracking/fluent.py#L92

Reply:
SGTM!